### PR TITLE
Add BOM ConfigMaps to tkr-system if configured

### LINKED ISSUE #2891: TKr BOM ConfigMaps are missing the tkr-system namespace
**Bug description**

TKR BOM ConfigMaps, as well as the bom-metadata ConfigMap should be available in the tkr-system namespace: they are expected to be there by legacy-style Addon Management and Tanzu CLI commands.

**Affected product area (please put an X in all that apply)**

**Expected behavior**

config maps found in the tkr-system namespace

**Steps to reproduce the bug**

deploy MC with pblcm feature flag.

**Version** (include the SHA if the version is not obvious)

**Environment where the bug was observed (cloud, OS, etc)**

**Relevant Debug Output (Logs, manifests, etc)**

### PR DESCRIPTION
**Pull request by imikushin**
### What this PR does / why we need it

New namespace for all things TKG is `tkg-system`. This includes TKR BOM ConfigMaps. Legacy style cluster operations in the tanzu CLI expect these in `tkr-system` though. This change makes sure these ConfigMaps get saved to both the new and legacy namespaces (if configured).

### Which issue(s) this PR fixes

Fixes #2891

### Describe testing done for PR

Manual testing done: BOM ConfigMaps appear in both `tag-system` and `tkr-system` when `--legacy-namespace=tkr-system` is set on tkr-source-controller.

### Release note

```release-note
NONE
```

### CODE REVIEW

File: packages/tkr-source-controller/bundle/config/values.yaml [hunk 1]
```diff
@@ -2,6 +2,7 @@
 #@overlay/match-child-defaults missing_ok=True
 ---
 namespace: tkg-system
+legacyNamespace: tkr-system
```
Comment (prkalle) [inline packages/tkr-source-controller/bundle/config/values.yaml:5]: I think this line should also be added in tkg meta package's `values.xml`

**Review by prkalle — COMMENTED** [review]
The changes look good to me with a minor comment. Thanks.

File: packages/tkr-source-controller/bundle/config/values.yaml (hunk 1 above)
Comment (imikushin) [reply packages/tkr-source-controller/bundle/config/values.yaml:5]: Good catch! Thanks @prkalle

**Review by wjun — DISMISSED** [review]
Please use a local branch in order the capa/capz public cloud pipeline tests get tested.

File: packages/tkr-source-controller/bundle/config/values.yaml (hunk 1 above)
Comment (imikushin) [reply packages/tkr-source-controller/bundle/config/values.yaml:5]: Done.

**Comment by imikushin** [comment]
> Please use a local branch in order the capa/capz public cloud pipeline tests get tested.

@wjun I'm not sure I understand. Can you please explain what you mean by "local branch" and how exactly you want it to be used? I am already using a local branch to create this PR. Do you want a certain PR/branch to be tested with this change? If so, could you provide a link?

Also, kindly, let's not use "Request Changes" button if there are no specific changes to the code that you think need to be made.

**Review by prkalle — APPROVED** [review]
LGTM.
As @wjun mentioned you may want to raise the PR from `tanzu-framework` branch instead of branch from your fork( so that more comprehensive github workflows would be able to run).  Thanks!

**Comment by imikushin** [comment]
> As @wjun mentioned you may want to raise the PR from `tanzu-framework` branch instead of branch from your fork( so that more comprehensive github workflows would be able to run). Thanks!

Thanks @prkalle! I'll do that (and provide a link to that PR here), for the sake of running more extensive CI checks. Once they succeed, I'll close it, and merge this one.

**Comment by wjun** [comment]
> > Please use a local branch in order the capa/capz public cloud pipeline tests get tested.
> 
> @wjun I'm not sure I understand. Can you please explain what you mean by "local branch" and how exactly you want it to be used? I am already using a local branch to create this PR. Do you want a certain PR/branch to be tested with this change? If so, could you provide a link?
> 
> Also, kindly, let's not use "Request Changes" button if there are no specific changes to the code that you think need to be made.

Got it. I mean not use a forked repo if the pr will have influence to public cloud providers too.

**Pull request merged**